onmt/translate/translator.py

```diff
@@ -428,6 +452,79 @@ def translate(
                       codecs.open(self.dump_beam, 'w', 'utf-8'))
         return all_scores, all_predictions
 
+    def _batch_predictions(self, predictions, bos, pad):
```

Comment:
This name should explicitly refer to the alignment task I think. Also, it's not very clear what this function is for.

Reply:
This function aims to rebuild the tgt tensor from predictions.
The predictions is n_batch List of n_best list containing generated predictions ended with eos index.
To achieve that, we need to do padding and add bos index.
It's a function that kind similar to pad() in torchtext.Field.

Reply:
maybe naming this function as _align_pad_prediction ?

Reply:
Seems good to me.